pkg/cmd/cli/cmd/edit.go

```diff
@@ -187,23 +199,30 @@ func (o *EditOptions) RunEdit() error {
 		if err != nil {
 			return preservedFile(err, file, o.out)
 		}
-		if bytes.Equal(original, edited) {
+		// Compare content without comments
+		uncommentedOriginal, originalErr := yaml.ToJSON(original)
+		uncommentedEdited, editedErr := yaml.ToJSON(edited)
```

Comment:
Added private wrappers for this. I went ahead and added manual comments removal (what I had before) in case yaml.ToJSON errors. PTAL